openmetadata-ui/src/main/resources/ui/src/components/TableProfiler/Component/ProfilerSettingsModal.tsx

```diff
@@ -121,9 +124,15 @@ const ProfilerSettingsModal: React.FC<ProfilerSettingsModalProps> = ({
     return metricsOptions;
   }, [columns]);
 
-  const { partitionColumnOptions, isPartitionDisabled } = useMemo(() => {
+  const partitionIntervalType = Form.useWatch(['partitionIntervalType'], form);
```

Comment:
typeof `partitionIntervalType` can be inferred if we provide proper types to form